java-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/resolution/typesolvers/JavaParserTypeSolver.java

```diff
@@ -41,12 +41,15 @@
 
     private TypeSolver parent;
 
-    private Map<String, CompilationUnit> parsedFiles = new HashMap<String, CompilationUnit>();
+    private Map<String, Optional<CompilationUnit>> parsedFiles = new HashMap<>();
     private Map<String, List<CompilationUnit>> parsedDirectories = new HashMap<>();
-    private Map<String, ReferenceTypeDeclaration> foundTypes=new HashMap<>();
+    private Map<String, ReferenceTypeDeclaration> foundTypes = new HashMap<>();
+    
+    private boolean hasValidSrcDir = false;
 
     public JavaParserTypeSolver(File srcDir) {
         this.srcDir = srcDir;
+        hasValidSrcDir = srcDir.exists() && srcDir.isDirectory();
```

Comment:
You are right! I can revert this change but in my profiling it occurred that doing this check over and over again consumes around 20% of the total CPU resources while using JSS to solve symbols in the JSS test project located on a harddrive that is not an SSD. What are your thoughts?